.github/workflows/test.yml

```
@@ -0,0 +1,40 @@
+      run: yarn tsc --noEmit
+
+    - name: build
+      run: yarn build
+    
+    - name: Run tests
+      run: yarn test --coverage
+  
+    
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v3
```

actionlint (1.7.4)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)